app/assets/javascripts/discourse/app/static/prosemirror/extensions/mention.js

```diff
@@ -74,6 +61,150 @@ const extension = {
       }
     },
   },
+
+  plugins({ pmState: { Plugin, PluginKey } }) {
+    const plugin = new PluginKey("mention");
+
+    return new Plugin({
```

Comment:
@renato I'm noticing a lot of these plugins/rules are ending up like an xmas tree of indentation, especially with plugins:

```
{ 
  {
    {
      {
        {
```

Maybe we can think about at some point storing the plugins in separate files, or definining them in some different way, even defining a class in the same file and hoisting it out of the main `plugins()` method so it's a bit cleaner? Personally I am finding a lot of these prosemirror extension source files a bit of a slog to read through because of this.

Reply:
We're already doing so in some cases, but yeah we can always organize it in a clearer way. Splitting is not always welcome as it can add a lot of indirection, but for example the image-toolbar is a separate extension

The main issue with this extraction is that you'll notice that a lot of the dependency injection will still be necessary, because plugins and theme components can't access the ProseMirror packages directly and it's always a good idea to make our core extensions serve as examples for external customizations. This would be much simpler if (when?) we had a way to import these /static/ packages from TCs/plugins.

That said, feel free to extract or suggest anything that helps code clarity!